### PR TITLE
[Enhancement][Vectorized] add constexpr_loop_match

### LINKED ISSUE #10282: [Enhancement][Vectorized] add constexpr_loop_match
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

we can use `constexpr_loop_match` to convert runtime variable to match constexpr.

### Solution

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
# Proposed changes

Issue Number: close #10282

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exec/join/vhash_join_node.cpp [hunk 1]
```diff
@@ -509,11 +517,6 @@ struct ProcessHashTableProbe {
                 }
                 output_block->get_by_position(result_column_id).column =
                         std::move(new_filter_column);
-            } else if constexpr (JoinOpType::value == TJoinOp::RIGHT_OUTER_JOIN) {
```
Comment (HappenLee) [inline be/src/vec/exec/join/vhash_join_node.cpp:512]: why delete this code?

File: be/src/vec/functions/function_json.cpp [hunk 2]
```diff
@@ -407,18 +407,36 @@ struct JsonParser<'3'> {
     }
 };
 
+template <>
+struct JsonParser<'4'> {
+    // time
+    static void update_value(StringParser::ParseResult& result, rapidjson::Value& value,
+                             StringRef data, rapidjson::Document::AllocatorType& allocator) {
+        value.SetString(data.data + 1, data.size - 2, allocator);
```
Comment (HappenLee) [inline be/src/vec/functions/function_json.cpp:415]: some comment? what means `data + 1` and `size - 2`

File: be/src/vec/exec/join/vhash_join_node.cpp (hunk 1 above)
Comment (BiteTheDDDDt) [reply be/src/vec/exec/join/vhash_join_node.cpp:512]: > why delete this code?
`RIGHT_OUTER_JOIN` is toe to toe with `RIGHT_SEMI_JOIN/RIGHT_ANTI_JOIN` now.

File: be/src/vec/functions/function_json.cpp (hunk 2 above)
Comment (BiteTheDDDDt) [reply be/src/vec/functions/function_json.cpp:416]: done.

**Review by HappenLee — DISMISSED** [review]
LGTM

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**